test/functional/rpc_psbt.py

```diff
@@ -370,6 +370,13 @@ def test_psbt_input_keys(psbt_input, keys):
         joined_decoded = self.nodes[0].decodepsbt(joined)
         assert len(joined_decoded['inputs']) == 4 and len(joined_decoded['outputs']) == 2 and "final_scriptwitness" not in joined_decoded['inputs'][3] and "final_scriptSig" not in joined_decoded['inputs'][3]
 
+        # Check that joining shuffles the inputs and outputs
```

Comment:
nit, could improve comment like "10 attempts should be enough to get a shuffled join".

Reply:
Done